queries/highlights.scm

```diff
@@ -13,6 +13,33 @@
 ((symbol) @keyword
   (#any-of? @keyword "defwindow" "defwidget" "defvar" "defpoll" "deflisten" "for" "geometry" "children" "struts"))
 
+; Builtin widgets
+
+((symbol) @type
+  (#any-of? @type
+    "box"
+    "button"
+    "calendar"
+    "centerbox"
+    "checkbox"
+    "circular-progress"
+    "color-button"
+    "color-chooser"
+    "combo-box-text"
+    "eventbox"
+    "expander"
+    "graph"
+    "image"
+    "input"
+    "label"
+    "literal"
+    "overlay"
+    "progress"
+    "revealer"
+    "scale"
+    "scroll"
+    "transform"))
```

Comment:
```suggestion
(list . 
  ((symbol) @type
    (#any-of? @type
      "box"
      "button"
      "calendar"
      "centerbox"
      "checkbox"
      "circular-progress"
      "color-button"
      "color-chooser"
      "combo-box-text"
      "eventbox"
      "expander"
      "graph"
      "image"
      "input"
      "label"
      "literal"
      "overlay"
      "progress"
      "revealer"
      "scale"
      "scroll"
      "transform")))
```

Although I think there's a bug in tree-sitter with anchors and predicate groups (or groups in general, see: https://github.com/tree-sitter/tree-sitter/pull/2107), I think it would still be a good idea for the future (if the linked PR gets merged) to limit this to the first element of a list, as I think it should only be valid there shouldn't it, or are there `symbol`s in there which are valid at a different place as well?

Also I'm not sure if we should use  `#any-of?` outside of neovim instead of `#match?` (see main, I've merged the other loop-widget PR which refactored `#any-of?` back to `#match?`, as it's apparently a predicate that's only implemented in neovim AFAIK, (see https://github.com/Philipp-M/tree-sitter-yuck/commit/04ca3d505072e1a85a43e2f96db563f8a258db73#r102328087).
Unfortunately the different interpretations of highlight/tree-sitter queries between different editors have already diverged quite bit (e.g. helix, neovim and emacs)...

Reply:
Maybe use `@tag` here and `@variable` for all other symbols? I have to use yuck more to get a feeling what the symbols are beyond the first list element. (The parser is pretty much a translation of the internal yuck parser, which uses `symbol` for every identitfier (outside of expressions))

Reply:
> Also I'm not sure if we should use `#any-of?` outside of neovim instead of `#match?` (see main, I've merged the other loop-widget PR which refactored `#any-of?` back to `#match?`, as it's apparently a predicate that's only implemented in neovim AFAIK.

I didn't know that, I just copied the structure used for `@keyword`, but if `#match?` is better I can change it here too, the only "issue" is that it would be an awfully long string

Reply:
> the only "issue" is that it would be an awfully long string

Yeah it's not completely optimal (as seen e.g. in https://github.com/cstrahan/tree-sitter-nix/blob/1b69cf1fa92366eefbe6863c184e5d2ece5f187d/queries/highlights.scm#L21), and maybe there should be more "official" predicates in upstream, which e.g. cover this use case more precisely (as it's quite common I think). I also think that there should be some kind of "standard"/reference or good practices for ids and other stuff in tree-sitter. E.g. the behavior of highlighting is different in the tree-sitter rust lib and neovim, neovim overwrites all previous queries, while helix/tree-sitter-lib takes the first query match and stops searching for other queries that apply to the code in question. so this doesn't work there as everything is already catched by the first symbol query.

But yeah, I guess it's better to go with `#match?` in the meantime and use `#any-of?` in nvim-treesitter.

Reply:
Fair enough. I was completely unaware of this difference between standard treesitter and neovim's implementation, which is actually pretty annoying.

Reply:
FWIW: https://github.com/nvim-treesitter/nvim-treesitter/issues/3944 and https://github.com/neovim/neovim/issues/22495